tests/db_functions/datetime/test_now.py

```diff
@@ -47,3 +48,17 @@ def test_basic(self):
             ["How to Time Travel"],
             lambda a: a.title,
         )
+
+    def test_microseconds(self):
+        Article.objects.create(
+            title="How to Django",
+            text=lorem_ipsum,
+            written=timezone.now(),
+        )
+
+        now_string = (
+            Article.objects.annotate(now_string=Cast(Now(), TextField()))
+            .get()
+            .now_string
+        )
+        self.assertRegex(now_string, r"^.*\.\d{3,6}")
```

Comment:
Could be a little more precise by using exactly 3 or 6, not between 3 and 6
```suggestion
        self.assertRegex(now_string, r"^.*\.(\d{3}|\d{6})")
```